src/pymor/discretizations/iosys.py

```diff
@@ -0,0 +1,1702 @@
+                raise TypeError('Expected A to be NumpyMatrixOperator with dense matrix (force_dense == False).')
+            if not ((isinstance(self.E, NumpyMatrixOperator) and not self.E.sparse) or
+                    _is_like_identity_operator(self.E)):
+                raise TypeError('Expected E to be NumpyMatrixOperator with dense matrix or IdentityOperator'
+                                '(force_dense == False).')
+
+        A = to_matrix(self.A, format='dense')
+        E = None if _is_like_identity_operator(self.E) else to_matrix(self.E, format='dense')
+        return spla.eigvals(A, E)
+
+    def eval_tf(self, s):
+        """Evaluate the transfer function.
+
+        The transfer function at :math:`s` is
+
+        .. math::
+            C (s E - A)^{-1} B + D.
+
+        .. note::
+            We assume that either the number of inputs or the number of
+            outputs is much smaller than the order of the system.
+
+        Parameters
+        ----------
+        s
+            Complex number.
+
+        Returns
+        -------
+        tfs
+            Transfer function evaluated at the complex number `s`,
+            |NumPy array| of shape `(self.p, self.m)`.
+        """
+        A = self.A
+        B = self.B
+        C = self.C
+        D = self.D
+        E = self.E
+
+        sEmA = LincombOperator((E, A), (s, -1))
+        if self.m <= self.p:
+            tfs = C.apply(sEmA.apply_inverse(B.as_range_array())).to_numpy().T
+        else:
+            tfs = B.apply_adjoint(sEmA.apply_inverse_adjoint(C.as_source_array())).to_numpy().conj()
+        if not _is_like_zero_operator(D):
+            tfs += to_matrix(D, format='dense')
+        return tfs
+
+    def eval_dtf(self, s):
+        """Evaluate the derivative of the transfer function.
+
+        The derivative of the transfer function at :math:`s` is
+
+        .. math::
+            -C (s E - A)^{-1} E (s E - A)^{-1} B.
+
+        .. note::
+            We assume that either the number of inputs or the number of
+            outputs is much smaller than the order of the system.
+
+        Parameters
+        ----------
+        s
+            Complex number.
+
+        Returns
+        -------
+        dtfs
+            Derivative of transfer function evaluated at the complex
+            number `s`, |NumPy array| of shape `(self.p, self.m)`.
+        """
+        A = self.A
+        B = self.B
+        C = self.C
+        E = self.E
+
+        sEmA = LincombOperator((E, A), (s, -1))
+        if self.m <= self.p:
+            dtfs = -C.apply(sEmA.apply_inverse(E.apply(sEmA.apply_inverse(B.as_range_array())))).to_numpy().T
+        else:
+            dtfs = -B.apply_adjoint(sEmA.apply_inverse_adjoint(E.apply_adjoint(sEmA.apply_inverse_adjoint(
+                C.as_source_array())))).to_numpy().conj()
+        return dtfs
+
+    @defaults('default_solver_backend', qualname='pymor.discretizations.iosys.LTISystem._lyap_solver')
```

Comment:
Was there a reason why the dispatch between the different implementations happens here instead of some `solve_lyap` in `pymor.algorithms`?

Reply:
There used to be `pymor.algorithms.lyapunov` and `pymor.algorithms.riccati`, which were then moved to `pymor.bindings` and `LTISystem`: 1490efbc9aac82202d2bc9bc5080dd38eb8998a4
I guess the dispatch was moved to `LTISystem` since `solve_lyap` and `solve_ricc` were only used there.

Reply:
Sounds good to me. As far as I see, `_lyap_solver` is not doing anything specific for the `LTISystem` class. I guess the same holds for `_ricc_solver` in `reductors.bt`.

Reply:
I have created an issue #476.